lib/types/index.d.ts

```diff
@@ -584,14 +584,19 @@ export namespace SourceCode {
 
 // #endregion
 
+type JSSyntaxElement = {
+	type: string;
+	loc?: ESTree.SourceLocation | null | undefined;
+};
```

Comment:
```suggestion
export type JSSyntaxElement = {
	type: string;
	loc?: ESTree.SourceLocation | null | undefined;
};
```

I think it would be useful to export this type like we do with `JSRuleDefinition`. We could add a test to check that the type can be imported.